### DISCUSSION TITLE
REST API for Copilot Chat

### DISCUSSION THREAD

**Discussion by AlexLee-101** [community/community#131114] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

Hi everyone,

Currently, I use the GitHub Copilot Business.
I want to use the Copilot Chat with REST API but I did not find the REST API for Copilot Chat.

Could you please provide some guidance or documentation on how to achieve this?

Thank you!

**Comment by OCEANOFANYTHINGOFFICIAL** [comment]
Hello @AlexLee-101 ,

GitHub Copilot Chat does not have a publicly available REST API for direct integration as of now. It primarily operates within the GitHub Copilot extension in supported IDEs like Visual Studio Code.

If you are looking to enhance or integrate Copilot functionalities into your workflow or application, consider these approaches:

1. **GitHub Copilot Extension**: Utilize the Copilot capabilities within supported IDEs (like Visual Studio Code) where it is integrated.
   
2. **Feature Requests**: GitHub continuously evolves its products based on user feedback. You can suggest the addition of a REST API for Copilot Chat by visiting GitHub's [feedback and feature request page](https://support.github.com/contact/feedback).

3. **GitHub Developer Platform**: Explore other GitHub APIs and integrations available through the [GitHub Developer Platform](https://developer.github.com/). While Copilot Chat may not have a REST API, other GitHub APIs can be leveraged for various integrations.

For the latest updates or any changes regarding Copilot Chat API availability, keep an eye on GitHub's official announcements or documentation.

Feel free to ask more questions or share your specific use case for further assistance!

**Comment by Akash1134** [comment, marked as answer]
Hi @AlexLee-101 

Copilot Chat does not have a REST API available for direct integration. It is designed to be used within the GitHub.com interface, providing assistance through chat embedded in the GitHub web interface.

Thanks!

**Reply by AlexLee-101** [reply]
@Akash1134 

Thank you for reply.
I created the https://github.com/orgs/community/discussions/131122 to request the REST API.
